File: Data Structures & Algorithms/invert-a-binary-tree/submission-3.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution {
    public TreeNode InvertTree(TreeNode root) {
        if (root == null) return null; 

        Queue<TreeNode> speicher = new Queue<TreeNode>();
        speicher.Enqueue(root);
        while (speicher.Count > 0)
        {        
            TreeNode node = speicher.Dequeue();
            TreeNode temp = node.left;
            node.left = node.right;
            node.right = temp;
            if( node.left != null) speicher.Enqueue(node.left);
            if( node.right!= null) speicher.Enqueue(node.right);
        }
        return root; 

    }
}

File: Data Structures & Algorithms/same-binary-tree/submission-3.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
        queue = deque()
        queue.append([p,q])

        while queue:
            nodep, nodeq = queue.popleft()
            
            if nodep is None and nodeq is None:
                continue
            if nodep is None or nodeq is None or nodep.val != nodeq.val:
                return False

            queue.append((nodep.left, nodeq.left))
            queue.append((nodep.right, nodeq.right))

        return True




